src/pvrclient-nextpvr.cpp

```diff
@@ -686,13 +686,22 @@ PVR_ERROR cPVRClientNextPVR::GetEpg(ADDON_HANDLE handle, int iChannelUid, time_t
         broadcast.strWriter           = NULL; // unused
         broadcast.strIMDBNumber       = NULL; // unused
 
-        // artwork URL
-        char artworkPath[128];
-        artworkPath[0] = '\0';
         if (g_bDownloadGuideArtwork)
         {
-          snprintf(artworkPath, sizeof(artworkPath), "http://%s:%d/service?method=channel.show.artwork&sid=%s&event_id=%d", g_szHostname.c_str(), g_iPort, m_sid, epgOid);
-          broadcast.strIconPath         = artworkPath;
+          // artwork URL
+          char artworkPath[512];
+          if (m_backendVersion < 50000)
+            snprintf(artworkPath, sizeof(artworkPath), "http://%s:%d/service?method=channel.show.artwork&sid=%s&event_id=%d", g_szHostname.c_str(), g_iPort, m_sid, epgOid);
+          else
+          {
+            if (g_sendSidWithMetadata)
+              snprintf(artworkPath, sizeof(artworkPath), "http://%s:%d/service?method=channel.show.artwork&sid=%s&name=%s", g_szHostname.c_str(), g_iPort, m_sid,UriEncode(title).c_str());
+            else
+              snprintf(artworkPath, sizeof(artworkPath), "http://%s:%d/service?method=channel.show.artwork&name=%s", g_szHostname.c_str(), g_iPort, UriEncode(title).c_str());
+            if (g_eventArtFormat == Portrait)
+              strcat(artworkPath, "&prefer=poster");
+          }
+          broadcast.strIconPath = artworkPath;
```

Comment:
Do you use StringUtils::Format else in the addon? Might be better to use that as you don't have to worry about running out of buffer space.

Just a suggestion mind you, it's up to you.

Reply:
Yes I am migrating away from snprintf That strcat worries me now since I guess there is a small possibility that someone might use a ridiculous hosthame.   The current code would crash in that case too.  I will change this to  ::Format()